bentoml/cli/__init__.py

```diff
@@ -74,8 +86,47 @@ def bentoml_cli(ctx, verbose, quiet):
     )
     @click.argument("api-name", type=click.STRING)
     @conditional_argument(archive_path is None, "archive-path", type=click.STRING)
+    @click.option(
+        '--with-conda',
+        is_flag=True,
+        default=False,
+        help="Run API server on a Conda environment.",
+    )
     @click.pass_context
-    def run(ctx, api_name, archive_path=archive_path):
+    def run(ctx, api_name, archive_path=archive_path, with_conda=False):
+        if with_conda:
+            config = load_bentoml_config(archive_path)
+            metadata = config['metadata']
+            env_name = metadata['service_name'] + '_' + metadata['service_version']
+
+            yaml = YAML()
+            yaml.default_flow_style = False
+            env_path = os.path.join('/tmp', env_name + '.yaml')
+            yaml.dump(config['env']['conda_env'], Path(env_path))
+
+            pip_req = os.path.join(archive_path, 'requirements.txt')
+
+            subprocess.call(
+                'command -v conda >/dev/null 2>&1 || {{ echo >&2 "--with-conda '
+                'parameter requires conda but it\'s not installed."; exit 1; }} && '
+                'conda env update -n {env_name} -f {env_file} && '
+                'conda init bash && '
```

Comment:
instead of init bash and activate, let's just do:
```
/Users/.../anaconda3/envs/{env_name...}/bin/bentoml {api_name} .... 
```

should work the same right?

Reply:
then it's difficult to handle those pip packages?

Reply:
you can do the same thing with pip right?  `/Users/.../anaconda3/envs/{env_name...}/bin/pip install ...`

Reply:
actually, let's keep this.. so if the user has set a different default env dir for conda, everything still works.

Reply:
let's just use `conda activate` here -  do we need `conda init bash` and `eval "$(conda shell.bash hook)"` tho? it seems unnecessary if we already require the user to install conda and have `conda` command available right?

Reply:
The reason is `conda` has some issues when running in a subshell.
I found the solution here https://github.com/conda/conda/issues/7980#issuecomment-492784093

Reply:
👍